test/dlc_tests/benchmark/sagemaker/huggingface/inference/test_performance_huggingface_inference.py

```diff
@@ -1,5 +1,177 @@
-import pytest
+# Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
 
+import pytest
+import sagemaker
+from sagemaker.model import Model
+from sagemaker.predictor import Predictor
+from sagemaker.serializers import JSONSerializer
+from sagemaker.deserializers import JSONDeserializer
+import time
+import numpy as np
+import signal
+import logging
 
-def dummy_test():
+class TimeoutException(Exception):
     pass
+
+def timeout_handler(signum, frame):
+    raise TimeoutException
+
+@pytest.mark.model("bert-base-uncased-pt")
+@pytest.mark.processor("cpu")
+@pytest.mark.cpu_test
+def test_sm_trained_model_cpu():
```

Comment:
Please use input `huggingface_pytorch_inference` to pass the URI for the HuggingFace PyTorch Inference image into this test, and add `cpu_only` to the inputs to only test CPU images.

The URI for the image that is built in a PR or a pipeline execution is automatically assigned to the variable `huggingface_pytorch_inference` whenever it is used as a parameter on any test function.

Please also change the name of this function to include `pytorch` in the name, since it is dedicated to PyTorch Inference images.

Also, please use pytest parametrization for tests that follow exactly the same workflow:
```suggestion
@pytest.mark.model("bert-base-uncased-pt")
@pytest.mark.processor("cpu")
@pytest.mark.cpu_test
@pytest.mark.parametrize("task,model", [
    ("fill-mask", BERT_BASE_UNCASED_PT_MODEL), ("fill-mask", ROBERTA_BASE_PT_MODEL)
])
def test_sm_trained_model_cpu():
```

Please also define the S3 URIs for images in variables common to all functions at the top of the script, so that they can be modified with a single code-change, in the future. Example:
```
BERT_BASE_UNCASED_PT_MODEL = "s3://aws-dlc-sample-models/huggingface_models/pytorch/bert-base-uncased-pytorch.tar.gz"
BERT_BASE_UNCASED_TF_MODEL = "s3://aws-dlc-sample-models/huggingface_models/tensorflow/bert-base-uncased-tensorflow.tar.gz"
ROBERTA_BASE_PT_MODEL = "s3://aws-dlc-sample-models/huggingface_models/pytorch/roberta-base-pytorch.tar.gz"
ROBERTA_BASE_TF_MODEL = "s3://aws-dlc-sample-models/huggingface_models/tensorflow/roberta-base-tensorflow.tar.gz"
```

This way we can have just 4 functions for different combinations of TF and PT + CPU and GPU.

Please also make sure that each test function has a different function name, as the test report would be generated incorrectly when two test functions share the same name in the same python test script.